lib/vagrant/util/downloader.rb

```diff
@@ -178,6 +191,47 @@ def head
 
       protected
 
+      # Apply any checksum validations based on provided
+      # options content
+      #
+      # @param source [String] Source of file
+      # @param path [String, Pathname] local file path
+      # @param checksums [Hash] User provided options
+      # @option checksums [String] :md5 Compare MD5 checksum
+      # @option checksums [String] :sha1 Compare SHA1 checksum
+      # @return [Boolean]
+      def validate_download!(source, path, checksums)
+        CHECKSUM_MAP.each do |type, klass|
+          if checksums[type]
```

Comment:
Should we add some kind of log message here, like:

> Validating checksum (md5) for X, expected: xxx actual: xxx

Could help in future debugging.